tests/test_vcf.py

```diff
@@ -325,3 +333,160 @@ def test_that__get_dataframe_uses_correct_columns_to_replace(
+                "POS",
+                "ID",
+                "REF",
+                "ALT",
+                "QUAL",
+                "FILTER",
+                "INFO",
+                "FORMAT",
+                "GENIE-SAGE-1-1",
+            ],
+            [[1, 2, 3, 4, 5, 6, 7, 8, 9, None]],
+            "vcf: Must not have missing values in GENIE-SAGE-1-1 column.\n",
+        ),
+        # Case 5: 11 columns (tumor-normal) — valid
+        (
+            [
+                "#CHROM",
+                "POS",
+                "ID",
+                "REF",
+                "ALT",
+                "QUAL",
+                "FILTER",
+                "INFO",
+                "FORMAT",
+                "GENIE-SAGE-1-1-normal",
+                "GENIE-SAGE-1-1-tumor",
+            ],
+            [[1, 2, 3, 4, 5, 6, 7, 8, 9, "a", "b"]],
+            "",  # no missing values
+        ),
+        # Case 6: 11 columns, tumor column has NA
+        (
+            [
+                "#CHROM",
+                "POS",
+                "ID",
+                "REF",
+                "ALT",
+                "QUAL",
+                "FILTER",
+                "INFO",
+                "FORMAT",
+                "GENIE-SAGE-1-1-normal",
+                "GENIE-SAGE-1-1-tumor",
+            ],
+            [[1, 2, 3, 4, 5, 6, 7, 8, 9, "a", None]],
+            "vcf: Must not have missing values in GENIE-SAGE-1-1-tumor column.\n",
+        ),
+        # Case 7: 11 columns, wrong normal/tumor naming
+        (
+            [
+                "#CHROM",
+                "POS",
+                "ID",
+                "REF",
+                "ALT",
+                "QUAL",
+                "FILTER",
+                "INFO",
+                "FORMAT",
+                "GENIE-SAGE-1-1-tumor",
+                "random_column",
+            ],
+            [[1, 2, 3, 4, 5, 6, 7, 8, 9, "x", "y"]],
+            "vcf: normal sample column must start with GENIE-SAGE\n",
+        ),
+        # Case 8: More than 11 columns
+        (
+            [
+                "#CHROM",
+                "POS",
+                "ID",
+                "REF",
+                "ALT",
+                "QUAL",
+                "FILTER",
+                "INFO",
+                "FORMAT",
+                "GENIE-SAGE-1-1-normal",
+                "GENIE-SAGE-1-1-tumor",
+                "extra_column",
+            ],
+            [[1, 2, 3, 4, 5, 6, 7, 8, 9, "x", "y", "z"]],
+            (
+                "vcf: Should not have more than 11 columns. Only "
+                "single sample or matched tumor normal vcf files are accepted.\n"
+            ),
+        ),
+    ],
+    ids=[
+        "only_nine_cols",
+        "valid_tumor_sample",
+        "invalid_tumor_sample_name",
+        "tumor_sample_col_has_nas",
+        "valid_matched_tumor_normal",
+        "invalid_normal_sample_name",
+        "more_than_11_cols",
+    ],
+)
+def test_validate_tumor_and_normal_sample_columns_exist(
```

Comment:
We really depend on the sites to have the tumor and normal columns in a specific order (1st tumor, then normal) and there's no other way to verify if one of the columns is tumor vs normal on just their column names. 

Generally based on past exp, sites using standard vcf files will always have the above, so this is OK for now